backend/app/user_handler.go

```diff
@@ -225,20 +251,47 @@ func (h *Handler) RegisterHandler(c *gin.Context) {
 
 		user.Mnemonic = mnemonic
 		user.StripeCustomerID = customer.ID
+		// Set user.Address from mnemonic
+		sponseeKeyPair, err := sr25519.Scheme{}.FromPhrase(mnemonic, "")
+		if err != nil {
+			log.Error().Err(err).Msg("failed to create keypair for SS58 address")
+			InternalServerError(c)
+			return
+		}
+		sponseeAddress, err := sponseeKeyPair.SS58Address(internal.SS58AddressFormat)
+		if err != nil {
+			log.Error().Err(err).Msg("failed to get SS58 address")
+			InternalServerError(c)
+			return
+		}
+		user.Address = sponseeAddress
 
 		err = h.db.RegisterUser(&user)
 		if err != nil {
 			log.Error().Err(err).Send()
 			InternalServerError(c)
 			return
 		}
+
+		if err := h.createKYCSponsorship(mnemonic); err != nil {
```

Comment:
u can pass the request's context to  `createKYCSponsorship` for better timeout handling